src/h5web/vis-packs/core/utils.ts

```diff
@@ -350,13 +343,20 @@ export function getValidDomainForScale(
   }
 
   const { min, max, positiveMin } = bounds;
-  if (scaleType !== ScaleType.Log || min * max > 0) {
+  const h5webScale = H5WEB_SCALES[scaleType];
+
+  if (h5webScale.supportsDomain([min, max])) {
     return [min, max];
   }
 
-  // Clamp domain minimum to first positive value,
-  // or return `undefined` if domain is not unsupported: `[-x, 0]`
-  return positiveMin !== Infinity ? [positiveMin, max] : undefined;
+  // Clamp domain minimum to first possible value
+  const [supportedMin] = h5webScale.supportedDomain;
+  const newMin = supportedMin > 0 ? positiveMin : supportedMin;
```

Comment:
Not really happy how this one turned out.

I want to use `positiveMin` in case of a log scale but `0` in case of a square root scale.

Reply:
Yeah, I feel like what needs changing is the condition used to find `positiveMin`, in the search for the domain. Here, you end up with `0` with the square root scale, even if the values don't contain `0`!